onnxruntime/contrib_ops/webgpu/quantization/subgroup_matrix_matmul_nbits.cc

```diff
@@ -426,6 +489,30 @@ Status ApplySubgroupMatrixMatMulNBits(const Tensor* a, const Tensor* b, const Te
                                       int32_t config_index,
                                       onnxruntime::webgpu::ComputeContext& context,
                                       Tensor* y) {
+  const auto& config = intel_supported_subgroup_matrix_configs[config_index];
+  const auto component_type = ComponentTypeName[static_cast<uint32_t>(std::get<2>(config))];
+  const auto m = std::get<4>(config);
+  const auto k = std::get<6>(config);
+
+  // Optimize the layout of input matrix A(MxK) for SubgroupMatrixLoad.
+  LayoutProgram layout_program{m, k, component_type};
```

Comment:
Because the apple shaders are not modified, I think you meant to run this only for the intel path. We should either update this code to support apple or put it behind a conditional check.

Reply:
@jchen10 Do we have test to cover this path? If no, please add one in case regressing existed path.